Modules/_datetimemodule.c

```diff
@@ -4839,6 +4852,41 @@ datetime_combine(PyObject *cls, PyObject *args, PyObject *kw)
     return result;
 }
 
+
+static PyObject *
+_sanitize_isoformat_str(PyObject *dtstr, unsigned char *needs_decref) {
+    // `fromisoformat` allows surrogate characters in exactly one position,
+    // the separator; to allow datetime_fromisoformat to make the simplifying
+    // assumption that all valid strings can be encoded in UTF-8, this function
+    // replaces any surrogate character separators with `T`.
+    Py_ssize_t len = PyUnicode_GET_LENGTH(dtstr);
```

Comment:
Using `PyUnicode_GET_LENGTH` requires having called `PyUnicode_READY` before. In this case (and below), just use `PyUnicode_GetLength()`.